app/src/main/java/com/oracle/javafx/scenebuilder/app/SceneBuilderApp.java

```diff
@@ -510,21 +529,76 @@ public void handleOpenFilesAction(List<String> files) {
 
         // Fix for #45
         if (userLibrary.isFirstExplorationCompleted()) {
-            performOpenFiles(fileObjs);
+            var openResult = performOpenFiles(fileObjs);
+            handleFileOpenResult(openResult, onSuccess);
         } else {
             // open files only after the first exploration has finished
             userLibrary.firstExplorationCompletedProperty().addListener(new InvalidationListener() {
                 @Override
                 public void invalidated(Observable observable) {
                     if (userLibrary.isFirstExplorationCompleted()) {
-                        performOpenFiles(fileObjs);
+                        var openResult = performOpenFiles(fileObjs);
                         userLibrary.firstExplorationCompletedProperty().removeListener(this);
+                        handleFileOpenResult(openResult, onSuccess);
                     }
                 }
             });
         }
     }
 
+    /**
+     * The onSuccess Runnable is only executed if the {@link FileOpenResult} has no
+     * errors at all.
+     * 
+     * @param result    {@link FileOpenResult} holding the list of files to be
+     *                  opened and a map of files with exceptions in case of an
+     *                  error.
+     * @param onSuccess A runnable holding the action to be performed on success.
+     */
+    private void handleFileOpenResult(FileOpenResult result, Runnable onSuccess) {
+        if (result.hasErrors()) {
+            showFileOpenErrors(result);
+        } else {
+            if (onSuccess != null) {
+                onSuccess.run();
+            }
+        }
+    }
+
+    /**
+     * In case of errors (>= 1 exception) a message dialog is shown to the user.
+     * There are versions for exactly 1 error and for more than 1 errors.
+     * 
+     * @param openResult {@link FileOpenResult} holding a map of files with
+     *                   exceptions.
+     */
+    private void showFileOpenErrors(FileOpenResult openResult) {
+        if (openResult.errors.size() == 1) {
```

Comment:
I'm not sure we get anything useful out of treating these as separate cases. Since we are making changes here, I think we can come up with a single general case for these, regardless of the number of errors. Maybe something along the lines of:

```
The following file(s) failed to open:
- file 1
- file 2
etc.
```

Reply:
Well, indeed for this PR it is not needed as SceneBuilder effectively cannot open multiple files. But there is another PR which enables Drag&Drop and there this becomes beneficial as then one can skim through the error messages.